Lib/test/test_capi/test_opt.py

```diff
@@ -1911,6 +1911,18 @@ def testfunc(n):
         self.assertNotIn("_COMPARE_OP_INT", uops)
         self.assertNotIn("_GUARD_IS_TRUE_POP", uops)
 
+    def test_call_len(self):
+        def testfunc(n):
+            a = [1, 2, 3, 4]
+            for _ in range(n):
+                _ = len(a) - 1
+
+        _, ex = self._run_with_optimizer(testfunc, TIER2_THRESHOLD)
```

Comment:
I don't know if it's needed in this case, but for other tests there's normally a 'control' variable that checks that the result is still correct even with the optimization. Something like this:
```suggestion
            x = 0
            a = [1, 2, 3, 4]
            for _ in range(n):
                _ = len(a) - 1
                if _ == 3:
                    x += 1

        _, ex = self._run_with_optimizer(testfunc, TIER2_THRESHOLD)
        self.assertEqual(res, TIER2_THRESHOLD)
```

Reply:
Good point, I'll do it.

Reply:
OK the problem is if I add this, _GUARD_TOS_INT and _GUARD_NOS_INT are appearing in the uops list.

Reply:
Yeah we can't do it. `x` is from outside the loop (trace), so the first `x += 1` *has* to have a guard somewhere.